Description: add the install target
 Noteworthy : Here are defined the headers for libv8-dev
Forwarded: not-needed
Author: Antonio Radici <antonio@dyne.org>
Last-Update: 2011-04-13

--- a/SConstruct
+++ b/SConstruct
@@ -1129,6 +1129,7 @@
   result.Add('mode', 'compilation mode (debug, release)', 'release')
   result.Add('sample', 'build sample (shell, process, lineprocessor)', '')
   result.Add('cache', 'directory to use for scons build cache', '')
+  result.Add('install', 'install result files', '')
   result.Add('env', 'override environment settings (NAME0:value0,NAME1:value1,...)', '')
   result.Add('importenv', 'import environment settings (NAME0,NAME1,...)', '')
   AddOptions(PLATFORM_OPTIONS, result)
@@ -1529,8 +1530,26 @@
   env.Alias('d8', d8s)
   env.Alias('preparser', preparsers)
 
+  install_dirs = []
+  destdir = ARGUMENTS.get('DESTDIR', '')
+  prefix = ARGUMENTS.get('PREFIX', '/usr')
+  libdir = [ destdir + prefix + '/lib/' ]
+  incdir = [ destdir + prefix + '/include/' ]
+  for lib in libraries:
+    env.Install(libdir, lib[0])
+    if dir not in install_dirs:
+      install_dirs += libdir
+  env.Install(incdir, [ 'include/v8.h', 'include/v8-debug.h', 'include/v8-profiler.h', 'include/v8-preparser.h', 'include/v8-testing.h', 'include/v8stdint.h' ]);
+  install_dirs += incdir
+  env.Alias('install', install_dirs)
+
+  # Always build at least the library
+  env.Default('library')
+
   if env['sample']:
     env.Default('sample')
+  elif env['install']:
+    env.Default('install')
   else:
     env.Default('library')
 
